packages/lib-classifier/src/plugins/drawingTools/models/marks/Mark/Mark.js

```diff
@@ -21,7 +21,11 @@ const BaseMark = types.model('BaseMark', {
     },
 
     get tool () {
-      return getParentOfType(self, Tool)
+      try {
+        return getParentOfType(self, Tool)
+      } catch (e) {
+        console.log(`no parent of ${Tool.type} available`)
+      }
```

Comment:
Should we enforce that marks must be created via `tool.createMark()`? I can't think of a situation where a mark would exist without a parent tool.

Reply:
Not in context of an app, but it does in the story!

Reply:
Surely the story should reflect how we're creating marks in the app? That's how I've been creating marks for tests.

Reply:
I put the example of how to do this in the ADR, which probably isn't the place anyone would look for an example.  : /
https://github.com/zooniverse/front-end-monorepo/blob/master/docs/arch/adr-22.md#the-drawing-tools-api

Reply:
By the way, this will attempt to find a parent of type `Tool`, but the transcription line's  parent tool is a `TranscriptionLineTool`. Each mark model has to implement this view for its specific tool type.

Reply:
So I'm not sure how to do that for the story without adding a Provider for MST stores so that the tree transversal function works. The ADR recommendation doesn't really apply here?

Reply:
This should work to create a line that you can pass into the React component.
```js
const transcriptionTool = TranscriptionLineTool.create({ type: 'transcriptionLine', color: 'red' })
const transcriptionLine - transcriptionTool.createMark({ id, x1, y1, x2, y2 })
```
`tool.createMark(snapshot)` should be described in the tools Readme, but I'm not sure how useful that is. PR's to improve the documentation are always welcome!

Maybe we need some examples like this? Examples would be useful for the classifier models too.

Reply:
I'm going to write some of the missing stories for the other drawing tools and see if my advice actually works.

Reply:
See #1413 for working Point and Line stories. I think this PR has forgotten to add a `tools` view to the `TranscriptionLine` model. By default, `mark.tool` will look for a parent Tool model, which will fail here because the tool model types don't match.

Reply:
#1414 will fix this by changing the fallback view to return the mark's grandparent, rather than search for a parent node of a specific type.